### PR TITLE
Split integration and unit tests

### PR DESCRIPTION
**Pull request by bruno-garcia**
Get tests to run without Unity

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <LangVersion>9</LangVersion>
     <Nullable>enable</Nullable>
-    <UnityVersion>2019.4.21f1</UnityVersion>
+    <UnityVersion>2019.4.19f1</UnityVersion>
```
Comment (semuserable) [inline src/Directory.Build.props:5]: Hm..

**Comment by semuserable** [comment]
You branch doesn't work for me. In order to not waste time (as it's still WIP on your side), I decided to give it a go on my own. I created `net48` runnable test project and put some simple test. It errors out during the run
image

It's strange because I'm reusing the same `D.B.p` as for other tests. Hm...

**Comment by semuserable** [comment]
Ok, resolved that by excluding `<Private>false</Private>`, but now another error.

image

**Comment by semuserable** [comment]
Ok, probably resolved that!

image

**Comment by semuserable** [comment]
So, the difference that I noticed, is we need to have a different `D.B.p` for `unit test` project with "privates" removed.

**Comment by semuserable** [comment]
WTF?

image

**Comment by semuserable** [comment]
This is so strange.

image

I thought it errors out because some Unity api is being called, but I used `UnityEngine.LogType` in the first test and it's totally ok with that! Why is it wrong for `UnityEventProcessor`? Hm..

**Comment by semuserable** [comment]
image

Ok, now it seems we're calling very specific Unity API and it doesn't like that. Let's find out!

**Comment by semuserable** [comment]
This call `Application.version` fails with `System.Security.SecurityException : ECall methods must be packaged into a system module.`

Investigation continued..

**Comment by semuserable** [comment]
`Time.realtimeSinceStartup` fails with `ECall`

```
[NativeProperty("Realtime")]
public static extern float realtimeSinceStartup { [MethodImpl(MethodImplOptions.InternalCall)] get; }
```

**Comment by semuserable** [comment]
`Application.installMode` fails with `ECall`

```csharp
public static extern ApplicationInstallMode installMode { [FreeFunction("GetApplicationInfo().GetInstallMode"), MethodImpl(MethodImplOptions.InternalCall)] get; }
```

**Comment by semuserable** [comment]
`Debug.isDebugBuild` fails with `ECall`

```csharp
[NativeProperty(TargetType = TargetType.Field)]
[StaticAccessor("GetBuildSettings()", StaticAccessorType.Dot)]
public static extern bool isDebugBuild { [MethodImpl(MethodImplOptions.InternalCall)] get; }
```

**Comment by semuserable** [comment]
As you can see, native `Unity` calls are not supported. If we want to test such logic in `unit test` manner, then we should remove all native calls.

What do you think? @bruno-garcia

File: src/Directory.Build.props (hunk 1 above)
Comment (bruno-garcia) [reply src/Directory.Build.props:5]: Only 22f1 shows up for install now for me and I don't have 21f1.

I'd revert this before mering.

**Comment by bruno-garcia** [comment]
> As you can see, native `Unity` calls are not supported. If we want to test such logic in `unit test` manner, then we should remove all native calls.
> 
> What do you think? @bruno-garcia

Thanks for investigating it further. Unity has a fork of Mono so it might not work at all even with Mono (which `dotnet test` on a `net48` on macOS run on).

Might be just too much work to get it to work. I want to take a stab a this before we give up though

**Comment by semuserable** [comment]
Noticed another strange thing. After opening up the solution, I've started to receive those weird errors

image

**Comment by bruno-garcia** [comment]
Seems like no chance to get this to work.

**Comment by bruno-garcia** [comment]
I mean for types that are pure C# (no dependency to Unity types) it works, but then I'm not sure it's worth adding.
We could consider if a lot of such types come in and things become too slow as they are

**Pull request closed without merging**